src/gui/inputdevices/PenInputHandler.cpp

```diff
@@ -120,16 +122,14 @@ auto PenInputHandler::actionStart(InputEvent const& event) -> bool {
             }
 
             xournal->selection->mouseDown(selType, selectionPos.x, selectionPos.y);
-            // Only modify selection and do not forward event to page
-            return true;
+            return isEventHandled;
         }
 
-
         xournal->view->clearSelection();
-        if (changeTool(event)) {
-            // Do not handle event in any further way to make click only deselect selection
-            return true;
-        }
+        changeTool(event);
+        // stop early to prevent drawing when clicking outside of the selection with the intention of deselecting
+        if (toolHandler->hasCapability(TOOL_CAP_SIZE))
```

Comment:
I assume you actually check here for whether the event originates from a pen or highlighter. If this is the case, this is definitiely the wrong method.Checking via side-effects is always bad since it breaks very easily as nobody expects this kind of usage. Instead check the toolType as is done in line 108.

Reply:
It's actually the Pen, Highlighter and Eraser :sweat_smile: 
But you're of course right. This was again the old habit of trading shortness for expressiveness :see_no_evil:. Will be changed.

Reply:
When working with the code the repetition of  `tool == TOOL_PEN || tool == TOOL_ERASER || tool == TOOL_HILIGHTER` seemed like to much code duplication for my taste, so I created a function `isDrawingTool` for `Tool` and a proxy for `ToolHandler`. That should give the best of both worlds :crossed_fingers: